internal/resource/mutation/mutation_test.go

```diff
@@ -230,6 +229,72 @@ func TestApply(t *testing.T) {
 				map[string]any{"name": 234},
 			}},
 		},
+		{
+			name:  "Complex_Nil",
+			path:  "self.spec.template.spec.containers[name='foo'].resources.limits.cpu",
+			value: nil,
+			obj: map[string]any{
+				"spec": map[string]any{
+					"selector": map[string]any{
+						"matchLabels": map[string]any{
+							"foo": "bar",
+						},
+					},
+					"template": map[string]any{
+						"metadata": map[string]any{
+							"labels": map[string]any{
+								"foo": "bar",
+							},
+						},
+						"spec": map[string]any{
+							"containers": []any{
```

Comment:
Worth testing multiple containers that match name='foo'? Or is that not a supported scenario?

Reply:
Overrides will work for fields that don't use `ownedByEno`. But the ownership check only works when matching against the key of the map slice since that's the key used in managedFields (docs coming soon on that lol). Since map slice keys are unique it isn't possible to have multiple matches